avocado/core/dispatcher.py

```diff
@@ -51,10 +51,18 @@ def plugin_type(self):
         else:
             return self.namespace
 
+    def fully_qualified_name(self, extension):
```

Comment:
OK, reading it a bit more, the `plugin_type` is actually dispatcher's method and the `fqn` is also influenced by that. I need to think about it a bit more, but currently the names are a bit misleading as they combine dispatcher and extension properties.